public/app/features/dashboard/dashboard_model.ts

```diff
@@ -146,6 +152,18 @@ export class DashboardModel {
       list: _.map(this.templating.list, variable => (variable.getSaveModel ? variable.getSaveModel() : variable)),
     };
 
+    if (!saveVariables && copy.templating.list.length === this.originalTemplating.length) {
+      for (let i = 0; i < copy.templating.list.length; i++) {
+        if (copy.templating.list[i].name === this.originalTemplating[i].name) {
+          copy.templating.list[i].current = this.originalTemplating[i].current;
+        }
+      }
+    }
+
+    if (!saveTimerange) {
```

Comment:
Tests and verification of saving/not saving time range is behaving as expected is needed.